packages/store/src/StoreCore.sol

```diff
@@ -797,44 +823,24 @@ library StoreCoreInternal {
     uint8 fieldIndex,
     uint256 byteLengthToPop
   ) internal {
-    uint8 dynamicSchemaIndex = fieldIndex - uint8(fieldLayout.numStaticFields());
-
-    // Load dynamic data length from storage
-    uint256 dynamicDataLengthSlot = _getDynamicDataLengthLocation(tableId, keyTuple);
-    PackedCounter encodedLengths = PackedCounter.wrap(Storage.load({ storagePointer: dynamicDataLengthSlot }));
-
-    // Update the encoded length
-    uint256 oldFieldLength = encodedLengths.atIndex(dynamicSchemaIndex);
-    encodedLengths = encodedLengths.setAtIndex(dynamicSchemaIndex, oldFieldLength - byteLengthToPop);
+    uint8 dynamicFieldIndex = fieldIndex - uint8(fieldLayout.numStaticFields());
 
-    // Set the new length
-    Storage.store({ storagePointer: dynamicDataLengthSlot, data: encodedLengths.unwrap() });
+    // Load the previous length of the field to set from storage to compute where to start to push
+    PackedCounter oldEncodedLengths = _loadEncodedDynamicDataLength(tableId, keyTuple);
+    uint40 oldFieldLength = uint40(oldEncodedLengths.atIndex(dynamicFieldIndex));
```

Comment:
I tend to prefer "previous" instead of "old" for these sorts of operations, feels a bit more specific/accurate.